src/controller.cpp

```diff
@@ -80,6 +80,13 @@ Controller::Controller() {
 
   connect(&m_handshakeTimer, &QTimer::timeout, this,
           &Controller::handshakeTimeout);
+
+  // Setup regular check for controller up state
+  connect(&m_gleanControllerUpTimer, &QTimer::timeout, this,
+          &Controller::periodicStateRecorder);
+  m_gleanControllerUpTimer.start(
```

Comment:
should we reset the time each time we go to `StateOn`/`StateOff` ? In order to have fix step data. Otherwise it could be this timer is triggered immediately after a connection/disconnection.

Reply:
I thought about this but didn't want to over complicate it. I think this will be fine actually for now and keeps the code much simpler. If it turns out analysis is complicated we can improve. But this was my motivation for putting this into the state change too so that we can use this instead of needing to also read from a state change event. So we can consider these events for usage metrics and state change events for troubleshooting our connection dramas.